examples/step_env.py

```diff
@@ -12,17 +13,16 @@
 args = parser.parse_args()
 
 # Construct the environment
-env = gym.make('MountainCar-v0')
+env = GarageEnv(gym.make('MountainCar-v0'))
```

Comment:
why not `GymEnv('MountainCar-v0')` and/or `GymEnv(gym.make('MountainCar-v0'))` (I think both should be supported)

Reply:
Yes it should be. I opened this draft PR to keep track of the design decisions in the description.The details here may likely change. But please do let me know if you have any questions/comments about the design decisions!